docs/diagrams/ResidentRoomAllocationDiagram.puml

```diff
@@ -0,0 +1,60 @@
+@startuml
+'https://plantuml.com/activity-diagram-beta
+
+start
+if (UniqueRoomList.contains()) then (true)
+    if (UniqueResidentList.contains()) then (true)
+        if (ResidentRoomList.isAllocated()) then (false)
+            if (ResidentRoomList.isRoomOccupied()) then (false)
+                if (ResidentRoomList.isResidentAllocated) then (false)
+                else (true)
+                    stop
+                endif
+                : Resident.setRoom();
+                : Room.occupy();
+                stop
+            else (true)
+                stop
+        endif
+        else (true)
+            stop
+    endif
+    else (false)
+        stop
+   endif
+
+else (false)
+    stop
+
+/'
```

Comment:
This looks like commented-out code. Was this intended to be here?

Reply:
removed this chunk of code. Thank for pointing out.